src/options.rs

```diff
@@ -137,24 +141,45 @@ impl Options {
   }
 
   pub(crate) fn bitcoin_rpc_client(&self) -> Result<Client> {
-    let cookie_file = self
-      .cookie_file()
-      .map_err(|err| anyhow!("failed to get cookie file path: {err}"))?;
-
     let rpc_url = self.rpc_url();
 
-    log::info!(
-      "Connecting to Bitcoin Core RPC server at {rpc_url} using credentials from `{}`",
-      cookie_file.display()
-    );
+    let config = self.load_config().unwrap();
+
+    let auth = if let (Some(rpc_user), Some(rpc_pass)) = (config.rpc_user, config.rpc_pass) {
```

Comment:
For context:

https://stackoverflow.com/questions/32272911/precedence-of-configuration-options-environment-registry-configuration-file-a